presto-hive/src/main/java/com/facebook/presto/hive/HivePageSourceProvider.java

```diff
@@ -447,6 +481,67 @@ protected static CacheQuota generateCacheQuota(HiveSplit hiveSplit)
             }
         }
 
+        return getPageSourceFromCursorProvider(
+                cursorProviders,
+                configuration,
+                session,
+                path,
+                start,
+                length,
+                fileSize,
+                storage,
+                effectivePredicate,
+                hiveColumns,
+                hiveStorageTimeZone,
+                typeManager,
+                tableName,
+                partitionKeyColumnHandles,
+                tableDataColumns,
+                tableParameters,
+                partitionDataColumnCount,
+                tableToPartitionMapping,
+                s3SelectPushdownEnabled,
+                remainingPredicate,
+                isPushdownFilterEnabled,
+                rowExpressionService,
+                customSplitInfo,
+                allColumns,
+                columnMappings,
+                outputIndices,
+                regularAndInterimColumnMappings,
+                bucketAdaptation);
+    }
+
+    private static Optional<ConnectorPageSource> getPageSourceFromCursorProvider(
```

Comment:
does making this non-static cut down any parameters? Just throwing it out there, given it seems pretty long